runtime/gc_glue_java/ObjectModelDelegate.hpp

```diff
@@ -315,6 +315,26 @@ class GC_ObjectModelDelegate
 		return J9GC_CLASS_IS_ARRAY(clazz);
 	}
 
+	/**
+	 * Get the preserved hashcode offset of forwarded header.
+	 *
+	 * @param[in]   Pointer to forwardedHeader the MM_ForwardedHeader instance encapsulating the object
+	 */
+	MMINLINE uintptr_t
+	getPreservedHashcodeOffset(MM_ForwardedHeader *forwardedHeader)
+	{
+		J9Class *clazz = _arrayObjectModel->getPreservedClass(forwardedHeader);
+		uintptr_t hashOffset;
+
+		if (_arrayObjectModel->isIndexable(clazz)) {
```

Comment:
hm, I don't like the looks of this method, afterall. 
2 times we ask arrayObjectModel something about an objects that we don't know yet it's an array. 
it works because those methods do not need any object model knowledge, but it just does not look good...
let's just remove this helper. anyhow, it would be just used once.